app/bundles/LeadBundle/Tests/Model/ImportModelTest.php

```diff
@@ -179,6 +181,41 @@ public function testStartImportWhenParallelLimitHit()
         $this->assertSame(Import::DELAYED, $entity->getStatus());
     }
 
+    public function testBeginImportWhenParallelLimitHit()
+    {
+        $model = $this->getMockBuilder(ImportModel::class)
+            ->setMethods(['checkParallelImportLimit', 'setGhostImportsAsFailed', 'saveEntity', 'getParallelImportLimit'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $model->method('checkParallelImportLimit')
+            ->will($this->returnValue(false));
+
+        $model->expects($this->once())
+            ->method('getParallelImportLimit')
+            ->will($this->returnValue(1));
+
+        $model->setTranslator($this->getTranslatorMock());
+
+        $entity = $this->initImportEntity(['canProceed']);
+
+        $entity->method('canProceed')
+            ->will($this->returnValue(true));
+
+        try {
+            $model->beginImport($entity, new Progress());
```

Comment:
Add `$this->fail()` to the next line to be sure that Exception was thrown.

Reply:
Added, thanks for this!